apps/kg/dataloader/KGDataset.py

```diff
@@ -59,17 +59,22 @@ class KGDataset:
     The triples are stored as 'head_name\trelation_name\ttail_name'.
     '''
     def __init__(self, entity_path, relation_path,
-                 train_path, valid_path=None, test_path=None,
-                 format=[0,1,2], read_triple=True, only_train=False,
-                 skip_first_line=False):
-        self.entity2id, self.n_entities = self.read_entity(entity_path)
-        self.relation2id, self.n_relations = self.read_relation(relation_path)
-
-        if read_triple == True:
+                 train_path, valid_path=None, test_path=None, local2global_path=None,
+                 format=[0,1,2], skip_first_line=False, partition=False, read_triple=True):
+        if partition == False:
+            self.entity2id, self.n_entities = self.read_entity(entity_path)
+            self.relation2id, self.n_relations = self.read_relation(relation_path)
             self.train = self.read_triple(train_path, "train", skip_first_line, format)
-            if only_train == False:
-                self.valid = self.read_triple(valid_path, "valid", skip_first_line, format)
-                self.test = self.read_triple(test_path, "test", skip_first_line, format)
+            self.valid = self.read_triple(valid_path, "valid", skip_first_line, format)
+            self.test = self.read_triple(test_path, "test", skip_first_line, format)
+        else:
+            with open(local2global_path) as f_ent:
+                self.n_entities = len(f_ent.readlines())
```

Comment:
does it requires to load all lines to get the number of lines? if it is, it'll take a long time to return if the graph is large. and all data read from the file are wasted.

Reply:
Yes. we just need to count of line here. I will change it.